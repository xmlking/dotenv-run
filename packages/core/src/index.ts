export * from './env';
export * from './plugin';