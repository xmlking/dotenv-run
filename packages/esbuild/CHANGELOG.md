# @dotenv-run/esbuild

## 1.2.0

### Minor Changes

- f63ecac: feat: add esbuild plugin

### Patch Changes

- Updated dependencies [f63ecac]
  - @dotenv-run/core@1.2.0
