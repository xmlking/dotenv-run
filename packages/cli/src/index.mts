#!/usr/bin/env node

import { filter, rootExpand } from "@dotenv-run/core";
import chalk from "chalk";
import minimist from "minimist";
import { run } from "./run.js";

const argv = minimist(process.argv.slice(2), {
  string: ["root", "env", "env-file"],
  boolean: ["silent"],
  alias: { help: "h", silent: "s", root: "r", env: "e", "env-file": "f" },
});

function help() {
  console.log(`
  Usage: dotenv-run [options] -- <command>
  
  Options:
  
    -h, --help     output usage information
    -e, --env      output usage information
    -s, --silent   do not print .env file paths
    -r, --root     root directory to search for .env files, defaults to current working directory
    
  Examples:
  
    dotenv-run -- npm start
    dotenv-run -r ../.. -- npm start
    dotenv-run -e prod -- npm start
  `);
}

if (argv.h) {
  help();
} else {
  const cmd = argv._[0];
  if (!cmd) {
    help();
    process.exit(1);
  }
  const envPaths = rootExpand(argv.r, argv.e, argv.f);
  import.meta.env = {};
  Object.keys(filter(process.env, /API/)).forEach((key) => {
    console.log(`${chalk.green("✔")} ${key}`);
    // import.meta.env[key] = process.env[key];
  });

  if (!argv.s) {
    envPaths.forEach((envPath) => {
      console.log(`${chalk.green("✔")} ${envPath}`);
    });
  }
  run(cmd, argv._.slice(1));
}
