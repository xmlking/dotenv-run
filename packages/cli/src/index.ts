#!/usr/bin/env node

import { expand, paths } from "@dotenv-run/core";
import * as chalk from "chalk";
import * as findUp from "find-up";
import * as minimist from "minimist";
import * as path from "path";
import { run } from "./run";

const argv = minimist(process.argv.slice(2), {
  string: ["root", "env"],
  boolean: ["silent"],
  alias: { help: "h", silent: "s", root: "r", env: "e" },
});

function help() {
  console.log(`
  Usage: dotenv-run [options] -- <command>
  
  Options:
  
    -h, --help     output usage information
    -e, --env      output usage information
    -s, --silent   do not print .env file paths
    -r, --root     root directory to search for .env files, defaults to current working directory
    
  Examples:
  
    dotenv-run -- npm start
    dotenv-run -r ../.. -- npm start
    dotenv-run -e prod -- npm start
  `);
}

if (argv.h) {
  help();
} else {
  const cmd = argv._[0];
  if (!cmd) {
    help();
    process.exit(1);
  }
  if (!argv.r) {
    let p = findUp.sync([
      "turbo.json",
      "nx.json",
      "lerna.json",
      "pnpm-workspace.yaml",
    ]);
    if (!p) p = findUp.sync(["package.json"]);
    argv.r = p ? path.dirname(p) : process.cwd();
  }
  const envPaths = paths(argv.e || process.env.NODE_ENV, argv.r);
  expand(envPaths);
  if (!argv.s) {
    envPaths.forEach((envPath) => {
      console.log(`${chalk.green("âœ”")} ${envPath}`);
    });
  }
  run(envPaths, cmd, argv._.slice(1));
}
