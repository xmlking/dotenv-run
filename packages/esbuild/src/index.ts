import { type DotenvRunOptions, env } from "@dotenv-run/core";
import type { Plugin } from "esbuild";

const dotenvRun = (options: DotenvRunOptions): Plugin => {
  return {
    name: "dotenv-run",
    setup(build) {
      const full = env(options).full;;
      const define = build.initialOptions.define ?? {};
      build.initialOptions.define = {
        ...full,
        ...define,
      };
    },
  };
};

export { dotenvRun, DotenvRunOptions };
export default dotenvRun;
