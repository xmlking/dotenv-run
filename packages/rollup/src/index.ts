import { DotenvRunOptions, env } from "@dotenv-run/core";
import replace from "@rollup/plugin-replace";

export default function (options: DotenvRunOptions) {
  const { full } = env(options);
  return {
    ...replace({
      preventAssignment: true,
      values: full,
    }),
  };
}

export { DotenvRunOptions };
