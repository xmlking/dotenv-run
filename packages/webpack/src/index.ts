import { DotenvRunOptions, plugin } from "@dotenv-run/core";
import * as webpack from "webpack";

export class DotenvRunPlugin {
  constructor(
    private options: DotenvRunOptions,
    private ssr = false
  ) {}
  apply(compiler: webpack.Compiler) {
    const { full, stringified } = plugin(this.options);
    const definePlugin = new webpack.DefinePlugin(
      this.ssr
        ? { ...full }
        : {
            "process.env": stringified,
            "import.meta.env": stringified,
            "import.meta.env.NODE_ENV": JSON.stringify(process.env.NODE_ENV),
          }
    );
    definePlugin.apply(compiler);
  }
}

export default DotenvRunPlugin;
