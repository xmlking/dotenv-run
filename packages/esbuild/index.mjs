import { build } from "esbuild";
import * as env from "./dist/cjs/index.js";

await build({
    write: true,
    bundle: true,
    entryPoints: [`test/app.js`],
    plugins: [
      env.dotenvRun({
        prefix: "MY_",
        verbose: true,
      }),
    ],
  });