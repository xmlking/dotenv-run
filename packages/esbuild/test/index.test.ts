import { build } from "esbuild";
import { describe, expect, it } from "vitest";
import { dotenvRun } from "../src/";
import { env } from "@dotenv-run/core";

describe("Usage with esbuild", () => {
  it("should replace environment variables using esbuild plugin", async () => {
    const results = await build({
      write: false,
      bundle: true,
      entryPoints: [`test/app.js`],
      plugins: [
        dotenvRun({
          prefix: "MY_",
          verbose: false,
        }),
      ],
    });

    expect(results.outputFiles.at(0)?.text).toMatchSnapshot();
  });

  it.skip("should replace environment variables using define option", async () => {
    const { full } = env({
      prefix: "MY_",
      verbose: false,
    });

    const results = await build({
      bundle: true,
      write: false,
      entryPoints: [`test/app.js`],
      define: full,
    });

    expect(results.outputFiles.at(0)?.text.trim()).toMatchSnapshot();
  });
});
