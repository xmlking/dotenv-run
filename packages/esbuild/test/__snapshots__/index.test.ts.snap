// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Usage with esbuild > should replace environment variables using esbuild plugin 1`] = `
"\\"use strict\\";
(() => {
  // test/util.js
  function meta() {
    return \\"https://dotenv-run.app\\";
  }

  // test/app.js
  console.log(\\"https://dotenv-run.app\\");
  console.log(meta());
})();
"
`;
